File: Frontend/src/app/models/approver/announcement.ts
import { SafeResourceUrl } from "@angular/platform-browser";
import { User } from "../admin/user";
import { AnnouncementFile } from "./announcementFile";

export class Announcement {
  id?: number;
  title?: string;
  message?: string;
  posted_by?: number;
  user?: User;
  audience?: string;
  created_date?: Date | null;
  expiry_date?: Date | null;
  image?: string;
  fileType?: string;         
  pdfUrl?: string;           
  safePdfUrl?: SafeResourceUrl;
  postedBy?: string;
  isFileEdited?: boolean;
  fileExsistance?: string;
  mainGuid?: string;
  previouseAnnouncementFile?: AnnouncementFile[];  
  announcementFile?: AnnouncementFile[];           
}

File: Backend/TaxManagementSystem/src/main/java/com/tms/Approver/Entity/Announcement.java
package com.tms.Approver.Entity;

import java.sql.Date;
import java.util.List;
import com.tms.Admin.Entity.User;
import com.tms.Maker.entity.TaxFile;

import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Announcement {
    private Long id;
    private String title;
    private String message;
    private Long posted_by;
    private User user;
    private String audience;
    private Date created_date;
    private Date expiry_date;
    private byte[] image;
    private String postedBy;
    private Boolean isFileEdited;
    private List<AnnouncementFile> previouseAnnouncementFile;
    private List<AnnouncementFile> announcementFile;
    private String fileExsistance;
    private String mainGuid;
}
